app/src/main/java/com/framgia/soundclound/screen/playlist/PlaylistModelView.java

```diff
@@ -0,0 +1,176 @@
+    private Context mContext;
+
+    public PlaylistModelView(Context context) {
+        mContext = context;
+        mPlaylistAdapter = new PlaylistAdapter(
+                AlbumRepository.getInstance(mContext).getAllAlbum());
+        mPlaylistAdapter.setOnItemAlbumClick(this);
+    }
+
+    @Bindable
+    public PlaylistAdapter getPlaylistAdapter() {
+        return mPlaylistAdapter;
+    }
+
+    public void setPlaylistAdapter(PlaylistAdapter playlistAdapter) {
+        mPlaylistAdapter = playlistAdapter;
+        notifyPropertyChanged(BR.playlistAdapter);
+    }
+
+    public void onFabAddAlbumClick() {
+        showInputDialog();
+    }
+
+    private void showInputDialog() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
+        builder.setTitle(R.string.title_add_album);
+        builder.setMessage(R.string.msg_name_album);
+        final EditText input = new EditText(builder.getContext());
+        builder.setView(input);
+        builder.setPositiveButton(R.string.action_add, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                String nameAlbum = input.getText().toString();
+                if (!nameAlbum.isEmpty()) {
+                    Album album = new Album();
+                    album.setName(nameAlbum);
+                    boolean result = AlbumRepository.getInstance(mContext).addAlbum(album);
+                    if (result) {
+                        mPlaylistAdapter.updateAlbums(
+                                AlbumRepository.getInstance(mContext).getAllAlbum());
+                    } else {
+                        dialog.cancel();
+                        Toast.makeText(
+                                mContext, R.string.msg_err_name_exist, Toast.LENGTH_SHORT).show();
+                    }
+                } else {
+                    Toast.makeText(
+                            mContext, R.string.msg_err_name_null, Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+        builder.setNegativeButton(R.string.action_cancel, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.cancel();
+            }
+        });
+        builder.show();
+    }
+
+
+    @Override
+    public void onItemClick(Album album) {
+        //// TODO: 10/1/2018  click item
+    }
+
+    @Override
+    public void onItemMoreClick(ImageView imageView, final Album album) {
+        if (album == null) {
+            return;
+        }
+        final AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
+        PopupMenu popupMenu = new PopupMenu(mContext, imageView);
+        popupMenu.getMenuInflater().inflate(R.menu.popup_menu_album, popupMenu.getMenu());
+        popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+            @Override
```

Comment:
WARNING: Method length is 72 lines (max allowed is 60).

Reply:
đã tách thành nhiều method  nhỏ hơn 60 lines nhưng CI vẫn bắt lỗi 
??????